app/models/custom_filter.rb

```diff
@@ -35,6 +35,13 @@ class CustomFilter < ApplicationRecord
   before_validation :clean_up_contexts
   after_commit :remove_cache
 
+  def expires_in
+    return @expires_in if defined?(@expires_in)
+    return nil if expires_at.nil?
+
+    [30.minutes, 1.hour, 6.hours, 12.hours, 1.day, 1.week].filter { |expires_in| expires_in.from_now >= expires_at }.first
```

Comment:
Use `find` instead of `filter` and `first` (also we prefer `select` to `filter` (they are the same thing))